### PR TITLE
Fb entrance page integration

### PR DESCRIPTION
**Pull request by guoz24**
## Description

- Old extraWebapp code completely removed (all html, css, JavaScript)
- extraWebapp directory now contains a React repository containing the new entrance page code
- gradle.build is edited to reflect new directory structure (it now copies from the build folder within the React repository) and to run "npm run build" before deploying

## Related PRs

https://github.com/RGLab/UITesting/pull/47

## Notion Ticket

https://www.notion.so/rglab/EP-Deploy-new-Entrance-Page-819dc1565bf14ee88c44c10d41081150

## PR Checklist for reviewer

- [x] Build it on your own local instance and test new features or changes.  
- [x] Pass UITesting on your local instance on the matching UITesting branch.  
- [x] It is functional and meets the requirements?   
- [x] Are any shorthand variable names are included in the [Glossary](https://www.notion.so/rglab/Glossary-of-variable-abbreviations-e205838b1f534abc903fa8c2228a6d7f)?  
- [x] Does it include descriptive code comments?   
- [x] Does it follow the [Style Guide](https://github.com/RGLab/LabKeyModules/#style-guide) for any languages included?  
- [x] Is it well-documented?
    * Module-specific README lists out any dependencies  
    * Sufficient instructions for setup  
    * Links to any relevant LabKey documentation   
- [x] Are any new features tested via UITesting or jest?  
- [x] Does it have any new dependencies? Is it justifiable?  
- [x] Is there code duplication that could be reduced?  
    * svg and style declarations to be moved to shared components library. https://www.notion.so/rglab/IRT-spring-cleaning-9323ba8e88cb424b84b48ccf73d5fcb6 
- [x] Are there performance improvements that could be made?

### CODE REVIEW

File: extraWebapp/build.gradle [hunk 1]
```diff
@@ -1,129 +1,21 @@
...
-
-if(moduleName == 'extraWebapp'){
-    deploy.doLast {
-        mkdir "${project.parent.deployDir}/extraWebapp"
-        copy {
-            from "."
-            exclude "build.gradle", "enlistment.properties"
-            into "${project.parent.deployDir}/extraWebapp"
-        }
-    }
```
Comment (juyeongkim) [inline extraWebapp/build.gradle:14]: Restore the build script and replace these with your codes

File: extraWebapp/rstudio-viewer.html [hunk 2]
```diff
@@ -1,261 +0,0 @@
-<script type='text/javascript' src='js/ext-4.2.1/ext-all-sandbox.js'></script>
```
Comment (juyeongkim) [inline extraWebapp/rstudio-viewer.html:1]: Restore this file and "deploy" (copy) this file into `${project.parent.deployDir}/extraWebapp` with `build.gradle` script. Or incorporate this in `/src`

File: .gitignore [hunk 3]
```diff
@@ -47,4 +47,5 @@ node_modules
 
 # Gradle
 .gradle/
-.envrc
\ No newline at end of file
+.envrc
+node_modules
```
Comment (juyeongkim) [inline .gitignore:51]: Delete the `build` directory and add it to `.gitignore`, so git won't be tracking and pushing them to GitHub.

File: extraWebapp/public/index.html [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <title>Welcome to ImmuneSpace</title>
+    <link rel="preconnect" href="https://fonts.gstatic.com/" crossorigin />
+    <link
+      href="https://fonts.googleapis.com/css?family=Muli:400,400i,500,700,700i,800&display=swap"
+      rel="stylesheet"
+    />
+
```
Comment (juyeongkim) [inline extraWebapp/public/index.html:14]: - Include Google Analytics script: https://github.com/RGLab/LabKeyModules/blob/master/extraWebapp/index.html#L24-L34
- Update `favicon.ico` and the logo png files.

**Review by juyeongkim — COMMENTED** [review]
I will build and test on my local machine once these issues are addressed.

File: extraWebapp/build/manifest.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
+  "short_name": "React App",
+  "name": "Create React App Sample",
```
Comment (juyeongkim) [inline extraWebapp/build/manifest.json:3]: Update these to "ImmuneSpace Entrance" and "Create ImmuneSpace Entrance Page"?

File: extraWebapp/rstudio-viewer.html (hunk 2 above)
Comment (guoz24) [reply extraWebapp/rstudio-viewer.html]: It's now in /public

**Review by helenmiller16 — CHANGES_REQUESTED** [review]
A few more comments! npm install failed on my local. At least part of it was based on not looking in the right place for @labkey/api. Make sure you change the React App defaults and remove the react logo, and use the new immunespace logo. Once I can build it on local, I'll go through and add any more notes.

**Comment by helenmiller16** [comment]
Can you also please include an overview of how it works and instructions for how to contribute in the README  in extraWebApp (and/or link to relevant Notion docs)

File: extraWebapp/src/js/login.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+//   } else {
+//     h.checked = false;
+//   }
+// }
+
+export function newLogin(setErrorMsg) {
+  const homePageUrl = window.location.href;
+
+  Ajax.request({
+    url: `${homePageUrl}login/home/loginApi.api`,
```
Comment (helenmiller16) [inline extraWebapp/src/js/login.js:26]: Is there a reason you need to insert the homePageUrl and not just use relative links like here: https://github.com/RGLab/LabKeyModules/blob/master/DataFinder/src/client/DataFinder/components/Banner.tsx#L193
That seems simpler to me, although I think the best would be to use the LabKey ActionURL API.

**Review by helenmiller16 — COMMENTED** [review]
Overall, everything is looking really good! Just trying to be somewhat consistent in code style. Make sure you're consistent in how you're declaring functions. And I noticed that you are passing in the full url to links, which I don't think should be necessary.

File: extraWebapp/src/js/login.js (hunk 6 above)
Comment (guoz24) [reply extraWebapp/src/js/login.js]: I did not think relative URLS would work for API calls but apparently it does? I've switched to relative.

**Review by helenmiller16 — APPROVED** [review]

**Pull request merged**